### PR TITLE
Additional build of distroless images

### PR DESCRIPTION
**Pull request by kramerul**
We did the following:
* We merged #12746, #12741, #12745, #12788, #12869 and #13395 into one PR
* We added build steps to istio-docker.mk for distroless images
* We are using the same Dockerfile for default and distroless images
* To build only the old images, you can use `make build docker docker.push BASE_DISTRIBUTIONS=default`
* Distroless images are push with a `-distroless` appended to `$TAG`

### CODE REVIEW

File: galley/docker/Dockerfile.galley [hunk 1]
```diff
@@ -1,3 +1,13 @@
-FROM istionightly/base_debug
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
+# -----------------------------------------------------------------------------
+FROM istionightly/base_debug as default
```
Comment (howardjohn) [inline galley/docker/Dockerfile.galley]: Does adding these FROMS make our image bigger (I think definitely no) or make our build times slower? Or is it effectively a NOP for the one we don't use.

If it does have an impact, maybe we can do `FROM $BASE` and have the arg be passed in as the full image?

Comment (howardjohn) [reply galley/docker/Dockerfile.galley]: Looking more I see the issue could be the default is not always the same

File: security/docker/Dockerfile.node-agent-k8s [hunk 2]
```diff
@@ -1,5 +1,14 @@
-FROM ubuntu:16.04
-ADD node_agent_k8s /
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
+# -----------------------------------------------------------------------------
+FROM ubuntu:16.04 as default
 RUN apt-get update && apt-get install -y ca-certificates
-RUN chmod 755 /node_agent_k8s
```
Comment (quanjielin) [inline security/docker/Dockerfile.node-agent-k8s]: why remove this line ? it was introduced for ingress-sds in https://github.com/istio/istio/commit/667b31b3c2b38674874ce24c603cf1847650424d#diff-db75d91f6316cdf9e537543775aa557e, @JimmyCYJ does remove this cause regression ?

Comment (JimmyCYJ) [reply security/docker/Dockerfile.node-agent-k8s]: Yes, when I added node agent as ingress gateway agent, I remember there was a permission issue without this line. Please keep it here.

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (jsPlachetka) [reply galley/docker/Dockerfile.galley]: It does not increase the image size. But make becomes slower. When both versions are build and pushed it is significantly slower. Nevertheless it is only an interim solution.

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (jsPlachetka) [reply security/docker/Dockerfile.node-agent-k8s]: Do you know the exact root cause for this issue? Is there a ticket or an integration test related to this problem?
There are many other Dockerfiles, where executables are simply `ADD`ed without running a chmod afterwards. It seems these Dockerfiles are working without this additional chmod.

Comment (JimmyCYJ) [reply security/docker/Dockerfile.node-agent-k8s]: I don't think there is a ticket. Without this line, if SDS is enabled at ingress gateway, the ingress gateway pod will have trouble to run, and kubectl describe pod <ingress gateway pod> shows there is a permission error for node agent. That is all I saw last time. I didn't find other useful information to catch the root cause. If keeping this line does not cause any critical problem, can we keep it? Otherwise, we need to verify removing this line does not break ingressgateway with SDS.

Comment (c0d1ngm0nk3y) [reply security/docker/Dockerfile.node-agent-k8s]: > ... If keeping this line does not cause any critical problem, can we keep it? ...

Not a critical problem, but `chmod` is not available on a distroless image per default. This is of course nothing that can't be fixed, but I think having a failing test case first would be essential. Just to see the problem and if it is fixed with chmod alone. Are you planning of such a SDS test?

**Comment by phil9909** [comment]
After rebasing the tests are green now. @howardjohn Could you have a look?

**Comment by howardjohn** [comment]
Sounds like SDS may still be broken by chmod?

**Comment by jsPlachetka** [comment]
With our last commit SDS should work again. We adjusted the istio-iptables.sh script to use dash instead of bash which is no longer available in our setting.

File: tests/istio.mk [hunk 3]
```diff
@@ -168,7 +168,7 @@ out_dir:
 test/local/auth/e2e_simple: out_dir generate_e2e_yaml
 	set -o pipefail; go test -v -timeout 25m ./tests/e2e/tests/simple -args --auth_enable=true \
 	--egress=false --ingress=false \
-	--rbac_enable=false --use_local_cluster --cluster_wide ${E2E_ARGS} ${T} ${EXTRA_E2E_ARGS} ${CAPTURE_LOG}
```
Comment (howardjohn) [inline tests/istio.mk]: Why is this changed?

Comment (kramerul) [reply tests/istio.mk]: This was changed by accident. We are using non local clusters for testing. Therefore, we changed this and forgot to revert our change.

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (howardjohn) [reply galley/docker/Dockerfile.galley]: What is the long term solution?

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (howardjohn) [reply security/docker/Dockerfile.node-agent-k8s]: I think this still needs to be fixed, we can't break nodeagent for everyone just for distroless. Agree it should have a test though

File: tools/packaging/common/istio-iptables.sh [hunk 4]
```diff
@@ -83,41 +87,39 @@ function isIPv4() {
...
-  for part in "${addr[@]}"; do
+  for part in ${addr}; do
+    part=${part%:}
     # check to not exceed number of parts in ipv6 address
-    if [[ ${number_of_parts} -ge 8 ]]; then
+    if [ ${number_of_parts} -ge 8 ]; then
         return 1
     fi
-    if [[ ${number_of_parts} -eq 0 ]] && ! [[ ${part} =~ ${ipv6section} ]]; then
+    if [ ${number_of_parts} -eq 0 ] && ! echo "${part}" | grep -q -E "${ipv6section}"; then
```
Comment (howardjohn) [inline tools/packaging/common/istio-iptables.sh]: I am pretty scared by all these changes to iptables. This is one of the most critical code paths for Istio, and it has virtually no code coverage. How confident are we that the script behaves the same now as it did before? Is there anything we can do to get more confidence?

Comment (c0d1ngm0nk3y) [reply tools/packaging/common/istio-iptables.sh]: > This is one of the most critical code paths for Istio, and it has virtually no code coverage. ...

How come? Sounds pretty scary to me too. If there is good you are afraid to change, that is never a good sign.

But a very good point though. We thought of rewriting `istio-iptables.sh` in go This could address the testing problem and gain confidence. What is your take on that?

Comment (howardjohn) [reply tools/packaging/common/istio-iptables.sh]: I am sure there is no *good* reason. In theory, we do have some coverage since all the E2E tests will make use of it. I think long term, a go rewrite is a good idea, we were recently discussing this in another PR that made some changes here.

I am not saying we should not make this change, but we should be cautious about it.

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (sdake) [reply security/docker/Dockerfile.node-agent-k8s]: Distroless or not, chmod should not be necessary.  No test cases fail with a lack of chmod AFAIK.  The reason chmod is unnecessary is docker will install the file with the permissions it was created with.  If the binary is 755, docker installs it as such.

In the interim, could you add chmod to the base image, create an issue to track the need to remove the chmod, and finally indicate the need to investigate why this particular binary needs chmod?  An alternative course of action would be to verify the system works without chmod (via a test case).
Cheers
-steve

File: tools/packaging/common/istio-iptables.sh (hunk 4 above)
Comment (sdake) [reply tools/packaging/common/istio-iptables.sh]: Please file an enhancement request to reimplement istio-iptables.sh in go.  We don't need that for 1.2, but it would be fantastic to get rid of bash which is very difficult to unit test.  Code that is difficult to automatically test slows down merge velocity for several reasons.

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (howardjohn) [reply security/docker/Dockerfile.node-agent-k8s]: I agree we should need to use chmod in the docker, we should make sure it is executable before adding it to the image (or if it already is, then do no action).

I just want to make sure we don't break what we have working now, even if the tests pass. I don't know if we have e2e tests for ingressgateway with sds so if we don't it makes sense the tests pass?

File: tools/packaging/common/istio-iptables.sh (hunk 4 above)
Comment (howardjohn) [reply tools/packaging/common/istio-iptables.sh]: https://github.com/istio/istio/issues/13967 created an issue

File: tools/packaging/common/istio-iptables.sh [hunk 5]
```diff
@@ -83,41 +87,39 @@ function isIPv4() {
...
         return 1
     fi
-    if ! [[ ${part} =~ ${ipv6section} ]]; then
-       if ! [[ "$part" == "" ]]; then
+    if ! echo "${part}" | grep -q -E "${ipv6section}"; then
```
Comment (sdake) [inline tools/packaging/common/istio-iptables.sh]: wow, this set of iptables changes will need some serious dedicated review time.  How about a middle ground where istio-iptables.sh stays as is, and a distroless ash version is created in the interim so the main stable images are not potentially destablized?

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (kramerul) [reply galley/docker/Dockerfile.galley]: The long term solution would be to remove the `default` section and only use the `distroless` version.

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (kramerul) [reply security/docker/Dockerfile.node-agent-k8s]: We tested the SDS case and found no issues with the `distroless` variant

File: mixer/docker/Dockerfile.mixer [hunk 6]
```diff
@@ -1,7 +1,17 @@
-FROM scratch
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
 
+# -----------------------------------------------------------------------------
+FROM scratch as default
 # obtained from debian ca-certs deb using fetch_cacerts.sh
```
[the remark this answers is not in the document]
Comment (kramerul) [reply mixer/docker/Dockerfile.mixer]: This is only needed for the `default` (old) variant of docker images. From line 10 to 11 you can find the variant for the distroless images

File: mixer/test/prometheus/cmd/Dockerfile [hunk 7]
```diff
@@ -12,7 +12,17 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM scratch
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
+# -----------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (kramerul) [reply mixer/test/prometheus/cmd/Dockerfile]: I will replace them

File: pilot/docker/Dockerfile.proxy_init [hunk 8]
```diff
@@ -1,8 +1,30 @@
-FROM ubuntu:xenial
+# BASE_DISTRIBUTION is used to switch between the old base distribution and distroless base images
+ARG BASE_DISTRIBUTION=default
+
+# -----------------------------------------------------------------------------
+FROM ubuntu:xenial as default
 RUN apt-get update && apt-get upgrade -y && apt-get install -y \
+    ash \
```
[the remark this answers is not in the document]
Comment (kramerul) [reply pilot/docker/Dockerfile.proxy_init]: Distroless uses busybox and busybox only provides ash and no bash. To use the same code for both variants (default,distroless), we decided to use ash.

File: tests/e2e/tests/pilot/listener_conflict_test.go [hunk 9]
```diff
@@ -219,25 +219,29 @@ func (i *pilotInfo) String() string {
 }
```
[the remark this answers is not in the document]
Comment (kramerul) [reply tests/e2e/tests/pilot/listener_conflict_test.go]: This change is required for this PR. Otherwise the test wont work for the distroless variant. 
I fear, if we split this up, it will take one additional week (not effort) to get this merged into master.
The reason for this change is that `curl` is no longer installed in the docker image. Therefore, we need to use a different (testing) pod to run the `curl` command.

File: tools/packaging/common/istio-iptables.sh [hunk 10]
```diff
@@ -1,4 +1,5 @@
-#!/bin/bash
+#!/bin/ash
```
[the remark this answers is not in the document]
Comment (kramerul) [reply tools/packaging/common/istio-iptables.sh]: Distroless doesn't support `bash`, even in the debug version.

We could reduce the risk by writing some addition tests for the refactored functions.

Another solution could be to rewrite only the functions `isValidIP`, `isIPv4` and `isIPv6` in go (including tests), build an executable and use this executable from the script.

File: tools/packaging/common/istio-iptables.sh (hunk 5 above)
Comment (kramerul) [reply tools/packaging/common/istio-iptables.sh]: This would mean that one of variants (distroless,default) would have a significant lower test coverage than the other.
If you would like to go with the distroless variant for 1.2, you need to increase the test coverage for the `distroless/ash` variant as soon as possible. 

In my point of view, the modified docker images introduce the same amount of risk. Therefore I would try to roll out this switch to `distroless` as soon as possible.

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (c0d1ngm0nk3y) [reply security/docker/Dockerfile.node-agent-k8s]: Shouldn't be the first action in such cases be to actually write a test to show that/if something is missing? If the passing tests are no indication that it works, what then?

**Comment by kramerul** [comment]
During testing, we found this issue: #13979, which is already fixed in the distroless variant.

File: tools/packaging/common/istio-iptables.sh (hunk 5 above)
Comment (kramerul) [reply tools/packaging/common/istio-iptables.sh]: We added a test under `tests/scripts/istio-iptables-test.sh` which verifies, that the new variant is doing the same as the old one (except the two bugs #13979 and #13980)

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (howardjohn) [reply galley/docker/Dockerfile.galley]: Am I understanding you correctly that you are saying, in the long term, we will only offer a distroless base and not a debian base?

I have not heard this before

Comment (kramerul) [reply galley/docker/Dockerfile.galley]: In my point of view, that would be the right direction. This was discussed under #12122.

Comment (howardjohn) [reply galley/docker/Dockerfile.galley]: I don't see any discussion of default in that issue. In all the WG meetings we have discussed how we want more options, not a different option, so I am not sure where you got the idea it would be the default from. The tools in the Debian image we provide are essential to debugging issues users face, and we use them daily.

I think everyone is really excited to have the option to use distroless, but not if it comes at the expense of te Debian one (which it doesn't have to, we can build any number of images)

Comment (kramerul) [reply galley/docker/Dockerfile.galley]: Sorry I was not part of the discussion. Perhaps, it was only my impression, that the direction should be a `distroless` based set of images.
I'm also fine with having both options in place.

**Comment by kramerul** [comment]
We created a PR (#14071) to use a go binary for istio-iptables.sh

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (howardjohn) [reply galley/docker/Dockerfile.galley]: I thought you were, hard to match a meeting voice to a github username sometimes though.

Thanks for pushing through all the pushback on the PR, this will be a great addition to Istio

**Comment by c0d1ngm0nk3y** [comment]
The idea is to keep the distro for istio-init for now and later tackle the `istio-iptables.sh`. We will update this PR soon...

File: tests/scripts/istio-iptables-test.sh [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+ip6tables -F,INPUT
+ip6tables -A,INPUT,-m,state,--state,ESTABLISHED,-j,ACCEPT
+ip6tables -A,INPUT,-i,lo,-d,::1,-j,ACCEPT
+ip6tables -A,INPUT,-j,REJECT
+END
+)
+
+assert_equals "$OUTPUT" "$EXPECTED_OUTPUT"
+
+echo "Test was sucessful"
```
[the remark this answers is not in the document]
Comment (mandarjog) [reply tests/scripts/istio-iptables-test.sh]: It appears to be testing implementation against an embedded golden file, nice !

Can you also add a test for `INBOUND_PORTS_INCLUDE=*`

File: tools/istio-docker.mk [hunk 12]
```diff
@@ -227,7 +227,9 @@ docker.node-agent-test: $(ISTIO_DOCKER)/node_agent.key
 # 4. This rule runs $(BUILD_PRE) prior to any docker build and only if specified as a dependency variable
 # 5. This rule finally runs docker build passing $(BUILD_ARGS) to docker if they are specified as a dependency variable
 
-DOCKER_RULE=time (mkdir -p $(DOCKER_BUILD_TOP)/$@ && cp -r $^ $(DOCKER_BUILD_TOP)/$@ && cd $(DOCKER_BUILD_TOP)/$@ && $(BUILD_PRE) docker build $(BUILD_ARGS) -t $(HUB)/$(subst docker.,,$@):$(TAG) -f Dockerfile$(suffix $@) .)
+BASE_DISTRIBUTIONS:=default distroless
+DEFAULT_DISTRIBUTION=default
+DOCKER_RULE=time (mkdir -p $(DOCKER_BUILD_TOP)/$@ && cp -r $^ $(DOCKER_BUILD_TOP)/$@ && cd $(DOCKER_BUILD_TOP)/$@ && $(BUILD_PRE) set -e && for distro in $(BASE_DISTRIBUTIONS); do tag=$(TAG)-$${distro}; docker build $(BUILD_ARGS) --build-arg BASE_DISTRIBUTION=$${distro} -t $(HUB)/$(subst docker.,,$@):$${tag%-$(DEFAULT_DISTRIBUTION)} -f Dockerfile$(suffix $@) . ; done )
```
[the remark this answers is not in the document]
Comment (kramerul) [reply tools/istio-docker.mk]: We calculate the time for each variant now.

File: galley/docker/Dockerfile.galley (hunk 1 above)
Comment (kramerul) [reply galley/docker/Dockerfile.galley]: Only the standard images will be built with default configuration.

File: Makefile [hunk 13]
```diff
@@ -231,15 +231,9 @@ ${ISTIO_BIN}/have_go_$(GO_VERSION_REQUIRED):
...
-# Ensure expected GOPATH setup
-.PHONY: check-tree
-check-tree:
-	@if [ ! "$(ISTIO_GO)" -ef "$(GO_TOP)/src/istio.io/istio" ]; then \
-		echo Not building in expected path \'GOPATH/src/istio.io/istio\'. Make sure to clone Istio into that path. Istio root="$(ISTIO_GO)", GO_TOP="$(GO_TOP)" ; \
-		exit 1; fi
 
 # Downloads envoy, based on the SHA defined in the base pilot Dockerfile
-init: check-tree check-go-version $(ISTIO_OUT)/istio_is_init
+init: check-go-version $(ISTIO_OUT)/istio_is_init
```
Comment (kramerul) [inline Makefile]: This check is no longer required with `go mod`

File: tests/e2e/tests/pilot/listener_conflict_test.go [hunk 14]
```diff
@@ -219,25 +219,29 @@ func (i *pilotInfo) String() string {
...
+	for i, podIP := range podIPs {
 		var err error
-		statuses[i], err = getPilotInfo(pod)
+		statuses[i], err = getPilotInfo(podIP, appPod)
 		if err != nil {
 			return nil, err
 		}
 	}
 	return statuses, nil
 }
```
[the remark this answers is not in the document]
Comment (kramerul) [reply tests/e2e/tests/pilot/listener_conflict_test.go]: Done.

File: pilot/docker/Dockerfile.proxyv2 [hunk 15]
```diff
@@ -22,23 +52,8 @@ ADD pilot-agent /usr/local/bin/pilot-agent
...
-ADD ca-certificates.tgz /
-
-RUN chmod 755 /usr/local/bin/envoy /usr/local/bin/pilot-agent
-
-# Sudoers used to allow tcpdump and other debug utilities.
-RUN useradd -m --uid 1337 istio-proxy && \
-    echo "istio-proxy ALL=NOPASSWD: ALL" >> /etc/sudoers && \
-    chown -R istio-proxy /var/lib/istio
+# istio-iptables.sh is not required for envoy
+# ADD istio-iptables.sh /usr/local/bin/istio-iptables.sh
```
Comment (kramerul) [inline pilot/docker/Dockerfile.proxyv2]: As far as we worked out, this script is never used by envoy. The container is not running in privileged mode. Therefore, running `iptables` commands should not be possible.
This is the reason, why we remove it from the docker image.
Could someone confirm?

Comment (howardjohn) [reply pilot/docker/Dockerfile.proxyv2]: I think there were plans/thoughts of making the init container use proxyv2 so the image is cached and there is less startup time on pods. I would guess this is why. I would say its safe to remove since we haven't done that, but I am not sure if there is anyone depending on the script being there and doing this themselves. @sdake any ideas?

Would there be an issue with having this in the distroless image? I get that it couldn't run it, but just so we don't break someone that is using it? Although I feel a bit like https://xkcd.com/1172/

**Comment by c0d1ngm0nk3y** [comment]
Is there any concern that we haven't addressed yet? The assumption is that this changes the status quo not at all, but opens the possibility to build&push distroless images to get some first experiences.

There are two main changes to not change the status quo:
1) The `init` container is not distroless, even in the distroless variant
2) The distroless variants are not built by default, but have to be enabled manually on the command line

File: tools/istio-docker.mk (hunk 12 above)
[the remark this answers is not in the document]
Comment (howardjohn) [reply tools/istio-docker.mk]: What is the difference though? Is it slower?

**Comment by howardjohn** [comment]
@c0d1ngm0nk3y so this is NOT blocked by https://github.com/istio/istio/pull/14071 because the init container doesn't use distroless yet, is that correct?

If so I think this is ok if you provide the times for building the docker images before/after this change (or do we just now build one and there is no difference?) and we add back the iptables script to proxvy2. I agree it may not be needed, but my understanding is that it is risk to keep it in there.
Thanks for all the work

@sdake any other comments?

File: tools/istio-docker.mk (hunk 12 above)
[the remark this answers is not in the document]
Comment (kramerul) [reply tools/istio-docker.mk]: There is no difference for the moment, because currently, we build only one variant.

**Comment by c0d1ngm0nk3y** [comment]
@howardjohn The #14071 is not needed for this. IT is not totally cleat how to proceed there, but we adapted this PR to be independent (by still using the script).

We can put the iptables script back in, but...
* Only for the default variant, right?
* Wouldn't it be good to find out why this is needed? We can quickly add it later, once we discovered that someone actually misses it. IMHO, that is better than putting it in "just to be safe". Your xkcd reference is quite fitting :)
* What use has this script on a non privileged container anyway?

**Comment by c0d1ngm0nk3y** [comment]
@howardjohn I measured the times of `make docker`:

```
real/user/system                       command

2m35/1m00/0m30        time GOOS=linux make docker (master branch)
2m36/1m00/0m30        time GOOS=linux make docker (distroless branch)
3m43/1m07/0m36        time GOOS=linux make docker DOCKER_BUILD_VARIANTS="default distroless"
2m50/1m01/0m31         time GOOS=linux make docker DOCKER_BUILD_VARIANTS="distroless"
```

Note: I executed each measurement only twice to have a verification, but I would take those numbers with a grain of salt.

**Comment by howardjohn** [comment]
@c0d1ngm0nk3y thanks for the times, great to see it is the same speed

LGTM, I'll give @sdake a chance to look as well since he had some comments, thanks for all the work I am excited to try it out

**Comment by howardjohn** [comment]
BTW test failures are unrelated, we are working on fixing the infra

File: pilot/docker/Dockerfile.proxyv2 (hunk 15 above)
Comment (sdake) [reply pilot/docker/Dockerfile.proxyv2]: Well, I wasn't aware of the choice to add init containers into the proxyv2 image itself.  As far as I know, the current plans indicate deprecating the init container entirely in favor of the CNI.  If the CNI cannot be used (@tiswanso are there envs where CNI won't work?), then we will need to have both models.

I still don't see a strong argument for keeping a file that is unused.  It can always be added back into the system (since this is master we are talking about) if someone has heartburn about it later...

**Comment by howardjohn** [comment]
/lgtm

@ozevren @nmittler needs galley approval

**Comment by ozevren** [comment]
Galley changes look ok to me.
/lgtm

File: Makefile (hunk 13 above)
Comment (ericvn) [reply Makefile]: We still don't use `go mod` for the build.

File: pilot/docker/Dockerfile.proxyv2 (hunk 15 above)
Comment (tiswanso) [reply pilot/docker/Dockerfile.proxyv2]: Yeah... the only way istio-iptables.sh would be usable would be if the container is started with net-admin.  So, for k8s that'd have to be some manual modification of the sidecar injection result.  I don't think that's likely.  For mesh expansion (VM/baremetal) I suppose it's possible there's some toolchain to set that up via the istio-proxy container.  IMO, its safe to remove.

Comment (howardjohn) [reply pilot/docker/Dockerfile.proxyv2]: fair enough, I am fine either way

**Comment by icygalz** [comment]
/test all

Test-infra is fixed.. triggering the test again.

**Comment by sdake** [comment]
I re-ran the circleci workflow.  If the gate is still failing, please forcepush a rebase so we can see the state of the gate.

Cheers
-steve

File: security/docker/Dockerfile.node-agent-k8s (hunk 2 above)
Comment (sushicw) [reply security/docker/Dockerfile.node-agent-k8s]: Ran into a problem caused by the missing chmod here, see https://github.com/istio/istio/issues/14718.
